ios/application_swift/RootViewController.swift

```diff
@@ -0,0 +1,51 @@
+import UIKit
+
+class RootViewController: UITableViewController {
+
+	var objects: [Date] = []
+
+	override func loadView() {
+		super.loadView()
+
+		tableView.register(UITableViewCell.self, forCellReuseIdentifier: "Cell")
+
+		title = "Root View Controller"
+		navigationItem.leftBarButtonItem = editButtonItem
+		navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(addButtonTapped))
+	}
+
+	@objc func addButtonTapped(_ sender: Any) {
+		objects.insert(Date(), at: 0)
+		tableView.insertRows(at: [IndexPath(row: 0, section: 0)], with: .automatic)
+	}
+
+	// MARK: - Table View Data Source
+
+	override func numberOfSections(in tableView: UITableView) -> Int {
+		return 1
+	}
+
+	override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+		return objects.count
+	}
+
+	override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+		let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath)
+
+		let date = objects[indexPath.row]
+		cell.textLabel!.text = date.description
```

Comment:
Would this be better if it were `cell.textLabel?.text` instead? cc @kirb

Reply:
Personally not a fan of using `?` in situations where it should be an error that the optional is nil. What does Apple’s master-detail app template do?

Reply:
Almost everything is force unwrapped in Apple's template, including `window` in AppDelegate.swift (whereas this template uses optional chaining for `window`)

Reply:
I’d make it all `!` then. Of course keep the optionality of the dictionary arg on application: didFinishLaunchingWithOptions: so it matches the UIApplicationDelegate protocol.